ocaml/idiomatic.ml

```diff
@@ -0,0 +1,73 @@
+type node = {
+  x : int;
+  y : int;
+  left : node option;
+  right : node option;
+}
+
+let rec merge lower greater =
+  match lower, greater with
+  | None, None -> None
+  | (Some _ as node), None | None, (Some _ as node) -> node
+  | Some lower, Some {y} when lower.y < y ->
+    Some {lower with right = merge lower.right greater}
+  | Some _, Some greater ->
+    Some {greater with left = merge lower greater.left}
+
+let empty = None
+
+let singleton value = {
+  x = value;
+  y = Random.int 0x3fffffff;
+  left = None;
+  right = None
+}
+
+let rec has_value tree value =
```

Comment:
This implementation does not match all the other implementation. Such benchmark is unfair.

Reply:
That's why I didn't make it the main version, but I didn't realize the benchmark algorithm is that much slower.